edb/server/dbview/dbview.pyx

```diff
@@ -887,6 +920,79 @@ cdef class DatabaseConnectionView:
         self._reset_tx_state()
         return side_effects
 
+    async def clear_cache_keys(self, conn) -> list[rpc.CompilationRequest]:
+        rows = await conn.sql_fetch(b'SELECT "edgedb"."_clear_query_cache"()')
+        rv = []
+        for row in rows:
+            query_req = rpc.CompilationRequest(
+                self.server.compilation_config_serializer
+            ).deserialize(row[0], "<unknown>")
+            rv.append(query_req)
+            self._db._eql_to_compiled.pop(query_req, None)
+        return rv
+
+    async def recompile_all(
+        self, conn, requests: typing.Iterable[rpc.CompilationRequest]
+    ):
+        compiler_pool = self.server.get_compiler_pool()
+        concurrency = max(1, compiler_pool.get_size_hint() - 1)
```

Comment:
This needs to be at most half of the compiler pool to avoid blocking live queries while cache refresh is being done.